### PR TITLE
PacketFence remote connector

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Adds a connector client and server to PacketFence so that a remote site can be controlled by a centralized PacketFence server

# Impacts
PFSSO, access reevaluation

# Delete branch after merge
YES

# NEWS file entries

TODO post-merge with the other big v12 items

### CODE REVIEW

**Comment by julsemaan** [comment]
@fdurand, do you want to do an initial code read before we meet to discuss this?

**Comment by fdurand** [comment]
> @fdurand, do you want to do an initial code read before we meet to discuss this?

yes i will read the code first.

**Comment by julsemaan** [comment]
> > @fdurand, do you want to do an initial code read before we meet to discuss this?
> 
> yes i will read the code first.

FYI, main areas of code that was changed in the chisel server:
https://github.com/inverse-inc/packetfence/blob/f46b620de07b6224931c6b5042269eb543ce1578/go/chisel/server/server_handler.go#L180-L186

https://github.com/inverse-inc/packetfence/blob/f46b620de07b6224931c6b5042269eb543ce1578/go/chisel/server/server_handler.go#L195-L297

https://github.com/inverse-inc/packetfence/blob/f46b620de07b6224931c6b5042269eb543ce1578/go/chisel/server/server.go#L180-L194

https://github.com/inverse-inc/packetfence/blob/f46b620de07b6224931c6b5042269eb543ce1578/go/chisel/server/server.go#L217-L226

The rest of the changes in chisel are related to making it listen on multiple hosts or opening a redis connection but the snippets above should be the core of the changes in chisel

File: addons/pfconnector/configure.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+echo "=================================================================="
+
+echo "Configuring connector with ID '$connector_id' and secret '$secret'"
+
+echo "AUTH=$connector_id:$secret" > /etc/pfconnector-client.env
+
+echo -n "Please enter the DNS name or IP address of the pfconnector server: "
+read connector_server
+
+echo "HOST=$connector_server" >> /etc/pfconnector-client.env
```
Comment (fdurand) [inline addons/pfconnector/configure.sh:35]: It´s where i think we can introduce the connector_ID.
Like if we do:
chisel client --host zammit https://1.2.3.4 3000

We will probably have the host header set to zammit.

**Comment by julsemaan** [comment]
@fdurand said it's OK to merge!

**Pull request merged**